media/src/main/java/edu/artic/media/audio/AudioPlayerService.kt

```diff
@@ -136,43 +137,51 @@ class AudioPlayerService : DaggerService() {
      * * [Playing]
      * * [Paused]
      * * [Stopped]
+     *
+     * To switch states, send a [PlayBackAction] to [audioControl].
      */
-    sealed class PlayBackState(val articAudioFile: ArticAudioFile) {
-        class Playing(articAudioFile: ArticAudioFile) : PlayBackState(articAudioFile)
-        class Paused(articAudioFile: ArticAudioFile) : PlayBackState(articAudioFile)
-        class Stopped(articAudioFile: ArticAudioFile) : PlayBackState(articAudioFile)
+    sealed class PlayBackState(val audio: AudioTranslation) {
+        class Playing(audio: AudioTranslation) : PlayBackState(audio)
+        class Paused(audio: AudioTranslation) : PlayBackState(audio)
+        class Stopped(audio: AudioTranslation) : PlayBackState(audio)
     }
 
     private val binder: Binder = AudioPlayerServiceBinder()
     private lateinit var playerNotificationManager: PlayerNotificationManager
+
+
     var articObject: ArticObject? = null
+        private set
+    var preferredAudio: AudioTranslation? = null
 
     private val audioControl: Subject<PlayBackAction> = BehaviorSubject.create()
     val audioPlayBackStatus: Subject<PlayBackState> = BehaviorSubject.create()
-    val currentTrack: Subject<ArticAudioFile> = BehaviorSubject.create()
+    val currentTrack: Subject<AudioTranslation> = BehaviorSubject.create()
 
     val disposeBag = DisposeBag()
 
     @Inject
     lateinit var analyticsTracker: AnalyticsTracker
+    @Inject
+    lateinit var languageSelector: LanguageSelector
 
     override fun onCreate() {
         super.onCreate()
         setUpNotificationManager()
         player.addListener(object : Player.DefaultEventListener() {
             override fun onPlayerStateChanged(playWhenReady: Boolean, playbackState: Int) {
-                val articAudioFile = articObject?.audioFile
+                val audio: AudioTranslation? = preferredAudio
```

Comment:
intermediate object not needed anymore. just a direct `preferredAudio?.let { given ->`

Reply:
Completed (with the other changes, it's now `(currentTrack as BehaviorSubject).value`)